File: HM/1_TwoSum.c
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {

int *array = malloc(2 * sizeof(int));
*returnSize = 2;

    for (int i = 0; i < numsSize; i++){
        for (int j = 0; j <numsSize; j++){
            if (nums[i] + nums[j] == target && i != j){
                array[0] = i;
                array[1] = j;
                return array;
            }
        }
    }
    return -1;
}

File: HM/1_TwoSum.py
def twoSum(nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(len(nums)):
                if nums[i] + nums[j] == target and i != j:
                    return [i, j]
                
###dict 방식으로 (Contains Duplicates2처럼 풀기)
def twoSum(self, nums: List[int], target: int) -> List[int]:
    d = dict()
    for i in range(len(nums)):
        diff = target - nums[i]
        if diff in d:
            return [d[diff], i]
        d[nums[i]] = i
